Packs/PAN-OS/Integrations/Panorama/Panorama.py

```diff
@@ -7011,6 +7154,1999 @@ def panorama_install_file_content_update_command(args: dict):
+    return result
+
+
+def get_routes(topology: Topology,
+               device_filter_string: str = None) -> ShowRoutingRouteCommandResult:
+    """
+    Pulls all route summary information from the topology
+    :param topology: `Topology` instance !no-auto-argument
+    :param device_filter_string: String to filter to only show specific hostnames or serial numbers.
+    """
+    result: ShowRoutingRouteCommandResult = FirewallCommand.get_routes(topology,
+                                                                       device_filter_string)
+    return result
+
+
+def get_system_info(topology: Topology,
+                    device_filter_string: str = None) -> ShowSystemInfoCommandResult:
+    """
+    Gets information from all PAN-OS systems in the topology.
+    :param topology: `Topology` instance !no-auto-argument
+    :param device_filter_string: String to filter to only show specific hostnames or serial numbers.
+    """
+    result: ShowSystemInfoCommandResult = UniversalCommand.get_system_info(topology,
+                                                                           device_filter_string)
+    return result
+
+
+def get_device_groups(topology: Topology, device_filter_string: str = None) -> list[DeviceGroupInformation]:
+    """
+    Gets the operational information of the device groups in the topology.
+    :param topology: `Topology` instance !no-auto-argument
+    :param device_filter_string: String to filter to only show specific hostnames or serial numbers.
+    """
+    result: list[DeviceGroupInformation] = PanoramaCommand.get_device_groups(topology, device_filter_string)
+    return result
+
+
+def get_template_stacks(topology: Topology, device_filter_string: str = None) -> list[
+    TemplateStackInformation]:
+    """
+    Gets the operational information of the template-stacks in the topology.
+    :param topology: `Topology` instance !no-auto-argument
+    :param device_filter_string: String to filter to only show specific hostnames or serial numbers.
+    """
+    result: list[TemplateStackInformation] = PanoramaCommand.get_template_stacks(topology,
+                                                                                 device_filter_string)
+    return result
+
+
+@dataclass
+class DemistoParameters:
+    """
+    Demisto Parameters
+    :param str hostnames: PAN-OS Hostnames (csv)
+    """
+    hostnames: str
+    credentials: dict
+    api_key: str = ""
+
+
+def convert_params_to_object(params: dict) -> DemistoParameters:
+    """Converts demisto params to the class used by NetOps code."""
+    server_url = params.get('server')
+    parsed_url = urlparse(server_url)
+    hostname = parsed_url.hostname
+    api_key = params.get('key', '')
+
+    return DemistoParameters(
+        hostnames=hostname,  # type: ignore
+        api_key=api_key,
+        credentials={}
+    )
+
+
+def run_test_command(func: Callable):
```

Comment:
Done

Reply:
I still don't see the change

Reply:
I changed the test function but I still seperated it next to the existing panorama_test command. I think it makes sense to test both aspects of this integration seperately, as puttinig the topology test under panorama_test function is misleading.

Reply:
this could be problematic, basically we should have a function that checks both of them and only if both of these functions return 'ok', then the test module should succeed in my opinion? could there be a possibility where we won't need to test_topology function while testing? if yes lets discuss that one